### PR TITLE
Client cleanup

### PR DESCRIPTION
**Pull request by LaserEyess**
Just some misc changes I made while looking into the slow upload issue. There were already two previous bugs in this file that were fixed by `g_clear_pointer()`, so I just changed the whole file to use that where relevant. I also removed `CURL_NO_SSL` as that's not really a relevant option these days. I doubt anyone actually builds libcurl wuthout some ssl library.

### CODE REVIEW

File: src/trg-client.c [hunk 1]
```diff
@@ -797,17 +780,15 @@ trg_response *dispatch_public_http(TrgClient *tc, trg_request *req) {
     g_free(cookie_header);
 
     if (headers)
-    	curl_slist_free_all(headers);
-
-    //g_message(response->raw);
+        curl_slist_free_all(headers);
```
Comment (TingPing) [inline src/trg-client.c:783]:
```c
g_clear_pointer (&headers, curl_slist_free_all);
```

File: src/trg-client.c [hunk 2]
```diff
@@ -864,13 +845,13 @@ dispatch_async(TrgClient * tc, JsonNode * req,
...
 gboolean async_http_request(TrgClient *tc, gchar *url, const gchar *cookie, GSourceFunc callback, gpointer data) {
-	trg_request *trg_req = g_new0(trg_request, 1);
-	trg_req->url = g_strdup(url);
+    trg_request *trg_req = g_new0(trg_request, 1);
+    trg_req->url = g_strdup(url);
 
-	if (cookie)
-		trg_req->cookie = g_strdup(cookie);
+    if (cookie)
+        trg_req->cookie = g_strdup(cookie);
```
Comment (TingPing) [inline src/trg-client.c:852]:
```c
trg_req->cookie = g_strdup(cookie);
```

`g_strdup(NULL)` returns `NULL`;

File: src/trg-client.c [hunk 3]
```diff
@@ -695,7 +681,7 @@ trg_http_perform_inner(TrgClient * tc, trg_request * request,
 
 
     if (headers)
-    	curl_slist_free_all(headers);
+        curl_slist_free_all(headers);
```
Comment (TingPing) [inline src/trg-client.c:684]: Another `g_clear_pointer()`

File: src/trg-client.c (hunk 1 above)
Comment (LaserEyess) [reply src/trg-client.c:783]: Hm, I also should have gotten rid of that comment

Comment (LaserEyess) [reply src/trg-client.c:783]: Oh, I did, that's a diff...

**Pull request merged**